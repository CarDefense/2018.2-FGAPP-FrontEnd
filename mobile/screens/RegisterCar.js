import React, { Component } from "react";
import { Constants, ImagePicker, Permissions } from 'expo';
import { TextField } from 'react-native-material-textfield';
import {
  ActivityIndicator,
  View,
  Text,
  TouchableOpacity,
  ScrollView,
  StyleSheet,
  Alert,
  Image,
  KeyboardAvoidingView
} from 'react-native';
import { CAR_API } from '../screens/tab_navigator/car_defense/screens/TabNavigator/const/Const'
import { Icon } from "native-base";


export default class RegisterCar extends Component {

  constructor(props) {
    super(props);

    this.plateRef = this.updateRef.bind(this, 'plate');
    this.modelRef = this.updateRef.bind(this, 'model');
    this.colorRef = this.updateRef.bind(this, 'color');
    this.onFocus = this.onFocus.bind(this);

    this.state = {
      plate: '',
      model: '',
      color: '',
      refreshing: false
    };
  }

  onFocus() {
    let { errors = {} } = this.state;

    for (let text in errors) {
      let ref = this[text];

      if (ref && ref.isFocused()) {
        delete errors[text];
      }
    }

    this.setState({ errors });
  }

  onChangeText(words) {
    ['plate']
      .map((text) => ({ text, ref: this[text] }))
      .forEach(({ text, ref }) => {
        if (ref && ref.isFocused()) {
          this.setState({ [text]: words });
        }
      });
    ['model']
      .map((text) => ({ text, ref: this[text] }))
      .forEach(({ text, ref }) => {
        if (ref && ref.isFocused()) {
          this.setState({ [text]: words });
        }
      });
    ['color']
      .map((text) => ({ text, ref: this[text] }))
      .forEach(({ text, ref }) => {
        if (ref && ref.isFocused()) {
          this.setState({ [text]: words });
        }
      });
  }

  async componentDidMount() {
    this.setState({ hasError: false, errorMessage: '' })
    const { state } = this.props.navigation;
    var id = state.params ? (state.params.user.id ? state.params.user.id : state.params.user.user_id) : undefined;

    let link = CAR_API + '/car/?token=' + id

    return fetch(link)
      .then((response) => response.json())
      .then((responseJson) => {

        this.setState({
          isLoading: false,
          dataSource: responseJson,
        }, function () {

        });

      })
      .catch((error) => {
        console.error(error);
      });
  }

  onPressButton = () => {
    let errors = {};
    let errorPlate = false;

    ['plate']
      .forEach((text) => {
        let value = this[text].value();

        if (!value) {
          errors[text] = 'Campo obrigatório.';
          errorPlate = true;
        } else {
          if ('plate' === text && value.length < 8) {
            errors[text] = 'Placa inválida. Ex.:AAA-0000';
            errorPlate = true;
          }
        }
      });

    ['plate']
      .forEach((text) => {
        let value = this[text].value();

        if (!value) {
          errors[text] = 'Campo obrigatório.';
          errorPlate = true;
        } else {
          if ('plate' === text && value.length < 8) {
            errors[text] = 'Placa inválida. Ex.:AAA-0000';
            errorPlate = true;
          }
        }
      });

    ['model']
      .forEach((text) => {
        let value = this[text].value();
      });

    ['color']
      .forEach((text) => {
        let value = this[text].value();
      });


    if (errorPlate == false) {

      const { state } = this.props.navigation;
      var id = state.params ? (state.params.user.id ? state.params.user.id : state.params.user.user_id) : undefined;

      const url = CAR_API + '/validate_car/' //cars db models url


      let car = JSON.stringify({
        id_token: id,
        plate: this.state.plate,
        model: this.state.model,
        color: this.state.color
      })

      console.log(car)

      fetch(url, {
        method: 'POST',
        headers: {
          'Accept': 'application/json',
          'Content-Type': 'application/json',
        },
        body: car
      }).then(response => { return response.json() }
      ).then(jsonResponse => {
        console.log(jsonResponse);
        // this.setState({ registered: true, registerMessage: 'Veículo cadastrado!' })
        Alert.alert('Veículo cadastrado!')
      }
      ).catch(error => {
        console.log(error)
        Alert.alert('Erro ao cadastrar!')
      })
    }
    this.setState({ errors });
  }

  updateRef(text, ref) {
    this[text] = ref;
  }

  render() {

    let { errors = {}, ...data } = this.state;
    let { plate = 'text' } = data;
    let { model = 'text' } = data;
    let { color = 'text' } = data;
    const { state } = this.props.navigation;
    var user = state.params ? (state.params.user.name ? state.params.user.name : state.params.user.username) : undefined;


    return (
<View style={{ backgroundColor: '#26C6DA', flex: 1 }}>
      <KeyboardAvoidingView behavior="position">
        <ScrollView>
          <View style={styles.container}>
            <View style={styles.header}>
              <View style={styles.headerContent}>
                <Image style={styles.avatar}
                source={require('../images/s.png')}
                />
                <Text style={styles.name}>Olá, {user}! Cadastre seus carros!</Text>
              </View>
            </View>
            <View style={styles.border}>
            <TextField
              ref={this.plateRef}
              value={data.plate}
              autoCorrect={false}
              enablesReturnKeyAutomatically={true}
              onFocus={this.onFocus}
              onChangeText={(plate) => this.setState({ plate })}
              // onChangeText={this.onChangeText}
              // onSubmitEditing={this.onSubmitPLate}
              returnKeyType='next'
              label='Placa'
              tintColor="white"
              underlineColorAndroid="transparent"
              maxLength={8}
              autoCapitalize="characters"
              error={errors.plate}
              textColor="white"
              labelPadding={5}
              inputContainerStyle = {{ marginHorizontal: 20 }}
            />
            <TextField
              ref={this.modelRef}
              value={data.model}
              autoCorrect={false}
              enablesReturnKeyAutomatically={true}
              onFocus={this.onFocus}
              onChangeText={(model) => this.setState({ model })}
              // onChangeText={this.onChangeText}
              // onSubmitEditing={this.onSubmitModel}
              returnKeyType='next'
              label='Modelo'
              tintColor="white"
              textColor="white"
              labelPadding={5}
              inputContainerStyle = {{ marginHorizontal: 20 }}
            />
            <TextField
              ref={this.colorRef}
              value={data.color}
              autoCorrect={false}
              enablesReturnKeyAutomatically={true}
              onFocus={this.onFocus}
              onChangeText={(color) => this.setState({ color })}
              // onChangeText={this.onChangeText}
              // onSubmitEditing={this.onSubmitMessage}
              returnKeyType='next'
              label='Cor'
              tintColor="white"
              textColor="white"
              labelPadding={5}
              inputContainerStyle = {{ marginHorizontal: 20 }}
            />
            <View style={styles.container1}>
              <TouchableOpacity
                color="#B2EBF2"
                onPress={this._takePhoto}
                containerViewStyle={{ width: '10%' }}
              >
                <Icon
                  type='FontAwesome'
                  name="camera"
                  style={{ color: "white" }}
                />
              </TouchableOpacity>

            </View>
            {this._maybeRenderImage()}
            {this._maybeRenderUploadingOverlay()}
            <View style={styles.container1}>
              <TouchableOpacity
                style={styles.button}
                color="white"
                onPress={this.onPressButton}
                containerViewStyle={{ width: '40%' }}
              >
                <Text style={{ color: '#B2EBF2', fontWeight: '800', fontSize: 15 }} >Cadastrar</Text>
              </TouchableOpacity>
              {/* {this.state.registered ? <Text style={{ flexDirection: 'row', justifyContent: 'center', color: '#5c68c3', marginTop: 20 }}>{this.state.registerMessage}</Text> : null } */}

            </View>
          </View>
          </View>
        </ScrollView>
      </KeyboardAvoidingView>
      </View>
    );
  }
  _maybeRenderUploadingOverlay = () => { //done
    if (this.state.uploading) {
      return (
        <View
          style={styles.maybeRenderUploading}>
          <ActivityIndicator color="#313869" size="large" />
        </View>
      );
    }
  };

  _maybeRenderImage = () => { //done
    let {
      image
    } = this.state;

    if (!image) {
      return;
    }

    // onPress={this._copyToClipboard}
    // onLongPress={this._share}

    return (
      <View
        style={styles.maybeRenderContainer}>
        <View
          style={styles.maybeRenderImageContainer}>
          <Image source={{ uri: image }} style={styles.maybeRenderImage} />
        </View>
      </View>
    );
  };

  _share = () => {
    Share.share({
      message: this.state.image,
      title: 'Check out this photo',
      url: this.state.image,
    });
  };

  _copyToClipboard = () => {
    Clipboard.setString(this.state.image);
    alert('Copied image URL to clipboard');
  };

  _takePhoto = async () => { //Done
    const {
      status: cameraPerm
    } = await Permissions.askAsync(Permissions.CAMERA);

    const {
      status: cameraRollPerm
    } = await Permissions.askAsync(Permissions.CAMERA_ROLL);

    // only if user allows permission to camera AND camera roll
    if (cameraPerm === 'granted' && cameraRollPerm === 'granted') {
      let pickerResult = await ImagePicker.launchCameraAsync({
        allowsEditing: true,
      });

      this._handleImagePicked(pickerResult);
    }
  };

  _pickImage = async () => {
    const {
      status: cameraRollPerm
    } = await Permissions.askAsync(Permissions.CAMERA_ROLL);

    // only if user allows permission to camera roll
    if (cameraRollPerm === 'granted') {
      let pickerResult = await ImagePicker.launchImageLibraryAsync({
        allowsEditing: true,

      });

      this._handleImagePicked(pickerResult);
    }
  };

  _handleImagePicked = async pickerResult => {
    let uploadResponse, uploadResult;

    try {
      this.setState({
        uploading: true
      });

      if (!pickerResult.cancelled) {
        uploadResponse = await uploadImageAsync(pickerResult.uri);
        uploadResult = await uploadResponse.json();

        console.log(uploadResult.image)

        this.setState({
          image: uploadResult.image
        });
      }
    } catch (e) {
      console.log({ uploadResponse });
      console.log({ uploadResult });
      console.log({ e });
      alert('Upload failed, sorry :(');
    } finally {
      this.setState({
        uploading: false
      });
    }
  };
}

async function uploadImageAsync(uri) {
  let apiUrl = CAR_API + '/documents/';

  // Note:
  // Uncomment this if you want to experiment with local server
  //
  // if (Constants.isDevice) {
  //   apiUrl = `https://your-ngrok-subdomain.ngrok.io/upload`;
  // } else {
  //   apiUrl = `http://localhost:3000/upload`
  // }

  let uriParts = uri.split('.');
  let fileType = uriParts[uriParts.length - 1];

  let formData = new FormData();
  formData.append('image', {
    uri,
    name: `photo.${fileType}`,
    type: `image/${fileType}`,
  });

  console.log(uri)

  let options = {
    method: 'POST',
    body: formData,
    headers: {
      Accept: 'application/json',
      'Content-Type': 'multipart/form-data',
    },
  };
  return fetch(apiUrl, options);
}

const styles = StyleSheet.create({
  container: {
    padding: 16,
    backgroundColor: '#26C6DA',
    flex: 1
  },
  container1: {
    marginTop: 13,
    flexDirection: 'row',
    justifyContent: 'center',
    bottom: 10
  },
  header2: {
    color: '#760f9f',
    textAlign: 'left',
    paddingLeft: 20,
    paddingTop: 35,
    fontWeight: '100',
    fontSize: 30,
  },
  button: {
    backgroundColor: "white",
    borderRadius: 15,
    height: 40,
    width: 120,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center'
  },
  item2: {
    alignItems: "center",
    backgroundColor: "#ffffff",
    flexGrow: 1,
    margin: 4,
    padding: 20,
    shadowColor: "#000000",
    shadowOpacity: 0.8,
    shadowRadius: 2,
    shadowOffset: {
      height: 1,
      width: 1
    },
    elevation: 4
  },
  text1: {
    color: "#760f9f",
    fontWeight: '200',
  },
  avatar: {
    width: 170,
    height: 170,
    borderRadius: 63,
    borderWidth: 7,
    borderColor: "white",
    marginBottom: 10,
  },
  name: {
    fontSize: 18,
    color: "white",
    fontWeight: '800',
  },
  headerContent: {
    padding: 30,
    alignItems: 'center',
  },
  box: {
    color: 'white'
  },
  header: {
    backgroundColor: "#26C6DA",
  },
  userInfo: {
    fontSize: 15,
    color: "white",
    fontWeight: '800',
    marginTop: 30,
    marginBottom: 15
  },
  border: {
    borderWidth: 5,
    borderRadius: 15,
    borderColor: "white",
    backgroundColor: '#B2EBF2'
  }
});

