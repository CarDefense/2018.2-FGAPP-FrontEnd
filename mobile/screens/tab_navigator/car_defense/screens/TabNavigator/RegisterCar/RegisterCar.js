import React, { Component } from "react";
import { TextField } from 'react-native-material-textfield';
import { View, Text, TouchableOpacity, ScrollView, StyleSheet, Alert, FlatList, RefreshControl } from 'react-native';
// import Expo from 'expo'
// import { Permissions, Notifications } from 'expo'
import jwt_decode from 'jwt-decode';

// var tk
// async function register() {
//   const { status } = await Expo.Permissions.askAsync(
//     Expo.Permissions.NOTIFICATIONS
//   );
//   if (status != 'granted') {
//     alert('You need to enable permissions in settings');
//     return;
//   }

//   const value = await Expo.Notifications.getExpoPushTokenAsync();
//   tk = value;
//   console.log(status, value);
// }

// async function list() {
//   const { status: existingStatus } = await Permissions.getAsync(
//     Permissions.NOTIFICATIONS
//   );
//   let finalStatus = existingStatus;

//   // only ask if permissions have not already been determined, because
//   // iOS won't necessarily prompt the user a second time.
//   if (existingStatus !== 'granted') {
//     // Android remote notification permissions are granted during the app
//     // install, so this will only ask on iOS
//     const { status } = await Permissions.askAsync(Permissions.NOTIFICATIONS);
//     finalStatus = status;
//   }

//   // Stop here if the user did not grant permissions
//   if (finalStatus !== 'granted') {
//     return;
//   }
//   var value
//   var tk = await Promise
//     .resolve(value = await Notifications.getExpoPushTokenAsync())
//     .then(x => value);
//   return tk;

// }


export default class RegisterCar extends Component {

  constructor(props) {
    super(props);

    this.plateRef = this.updateRef.bind(this, 'plate');
    this.modelRef = this.updateRef.bind(this, 'model');
    this.colorRef = this.updateRef.bind(this, 'color');
    this.onFocus = this.onFocus.bind(this);

    this.state = {
      plate: '',
      model: '',
      color: '',
      refreshing: false
    };
  }

  onFocus() {
    let { errors = {} } = this.state;

    for (let text in errors) {
      let ref = this[text];

      if (ref && ref.isFocused()) {
        delete errors[text];
      }
    }

    this.setState({ errors });
  }

  onChangeText(words) {
    ['plate']
      .map((text) => ({ text, ref: this[text] }))
      .forEach(({ text, ref }) => {
        if (ref && ref.isFocused()) {
          this.setState({ [text]: words });
        }
      });
    ['model']
      .map((text) => ({ text, ref: this[text] }))
      .forEach(({ text, ref }) => {
        if (ref && ref.isFocused()) {
          this.setState({ [text]: words });
        }
      });
    ['color']
      .map((text) => ({ text, ref: this[text] }))
      .forEach(({ text, ref }) => {
        if (ref && ref.isFocused()) {
          this.setState({ [text]: words });
        }
      });
  }

  async componentDidMount() {
    this.setState({ hasError: false, errorMessage: '' })
    const { state } = this.props.navigation;
    var token = state.params ? state.params.token : undefined;
    user = jwt_decode(token)
    let link = 'http://192.168.0.4:8003/car/?token=' + user.user_id

    return fetch(link)
      .then((response) => response.json())
      .then((responseJson) => {

        this.setState({
          isLoading: false,
          dataSource: responseJson,
        }, function () {

        });

      })
      .catch((error) => {
        console.error(error);
      });
  }

  _onRefresh = () => {
    this.setState({ refreshing: true });
    this.componentDidMount().then(() => {
      this.setState({ refreshing: false });
    });
  }


  onPressButton = () => {
    let errors = {};
    let errorPlate = false;

    ['plate']
      .forEach((text) => {
        let value = this[text].value();

        if (!value) {
          errors[text] = 'Campo obrigatório.';
          errorPlate = true;
        } else {
          if ('plate' === text && value.length < 8) {
            errors[text] = 'Placa inválida. Ex.:AAA-0000';
            errorPlate = true;
          }
        }
      });

    ['plate']
      .forEach((text) => {
        let value = this[text].value();

        if (!value) {
          errors[text] = 'Campo obrigatório.';
          errorPlate = true;
        } else {
          if ('plate' === text && value.length < 8) {
            errors[text] = 'Placa inválida. Ex.:AAA-0000';
            errorPlate = true;
          }
        }
      });

    ['model']
      .forEach((text) => {
        let value = this[text].value();
      });

    ['color']
      .forEach((text) => {
        let value = this[text].value();
      });


    if (errorPlate == false) {

      const { state } = this.props.navigation;
      var token = state.params ? state.params.token : undefined;
      user = jwt_decode(token)

      const url = 'http://192.168.0.4:8003/car/' //cars db models url

      let notification = JSON.stringify({
        id_token: user.user_id,
        plate: this.state.plate,
        model: this.state.model,
        color: this.state.color
      })

      console.log(notification)

      fetch(url, {
        method: 'POST',
        headers: {
          'Accept': 'application/json',
          'Content-Type': 'application/json',
        },
        body: notification
      }).then(response => { return response.json() }
      ).then(jsonResponse => {
        console.log(jsonResponse);
        // this.setState({ registered: true, registerMessage: 'Veículo cadastrado!' })
        Alert.alert('Veículo cadastrado!')
      }
      ).catch(error => {
        console.log(error)
        Alert.alert('Erro ao cadastrar!')
      })
    }
    this.setState({ errors });
  }

  updateRef(text, ref) {
    this[text] = ref;
  }

  render() {

    let { errors = {}, ...data } = this.state;
    let { plate = 'text' } = data;
    let { model = 'text' } = data;
    let { color = 'text' } = data;

    return (
      <ScrollView
        refreshControl={
          <RefreshControl
            refreshing={this.state.refreshing}
            onRefresh={this._onRefresh}
          />
        }
      >
        <View style={styles.container}>
          <Text style={styles.header}> Cadastrar carro</Text>
          <TextField
            ref={this.plateRef}
            value={data.plate}
            autoCorrect={false}
            enablesReturnKeyAutomatically={true}
            onFocus={this.onFocus}
            onChangeText={(plate) => this.setState({ plate })}
            // onChangeText={this.onChangeText}
            // onSubmitEditing={this.onSubmitPLate}
            returnKeyType='next'
            label='Placa'
            tintColor="#760f9f"
            underlineColorAndroid="transparent"
            maxLength={8}
            autoCapitalize="characters"
            error={errors.plate}
          />
          <TextField
            ref={this.modelRef}
            value={data.model}
            autoCorrect={false}
            enablesReturnKeyAutomatically={true}
            onFocus={this.onFocus}
            onChangeText={(model) => this.setState({ model })}
            // onChangeText={this.onChangeText}
            // onSubmitEditing={this.onSubmitModel}
            returnKeyType='next'
            label='Modelo'
            tintColor="#760f9f"
          />
          <TextField
            ref={this.colorRef}
            value={data.color}
            autoCorrect={false}
            enablesReturnKeyAutomatically={true}
            onFocus={this.onFocus}
            onChangeText={(color) => this.setState({ color })}
            // onChangeText={this.onChangeText}
            // onSubmitEditing={this.onSubmitMessage}
            returnKeyType='next'
            label='Cor'
            tintColor="#760f9f"
          />
        </View>
        <View style={styles.container1}>
          <TouchableOpacity
            style={styles.button}
            color="#5c68c3"
            onPress={this.onPressButton}
            containerViewStyle={{ width: '40%' }}
          >
            <Text style={{ color: 'white' }} >Cadastrar</Text>
          </TouchableOpacity>
          {/* {this.state.registered ? <Text style={{ flexDirection: 'row', justifyContent: 'center', color: '#5c68c3', marginTop: 20 }}>{this.state.registerMessage}</Text> : null } */}

        </View>
        <View style={styles.container1}>
          <FlatList
            data={this.state.dataSource}
            renderItem={({ item }) => {
              return (
                <View style={styles.item2}>
                  <Text style={styles.text1}>{item.plate}</Text>
                </View>

              );
            }}
            keyExtractor={({ id }, index) => id.toString()}
          />
        </View>
      </ScrollView>
    );
  }
}

const styles = StyleSheet.create({
  container: {},
  container1: {
    marginTop: 80,
    flexDirection: 'row',
    justifyContent: 'center'
  },
  header: {
    color: '#760f9f',
    textAlign: 'center',
    fontWeight: '200',
    fontSize: 50,
    marginTop: 25
  },
  header2: {
    color: '#760f9f',
    textAlign: 'left',
    paddingLeft: 20,
    paddingTop: 35,
    fontWeight: '100',
    fontSize: 30,
  },
  button: {
    backgroundColor: "#760f9f",
    borderRadius: 15,
    height: 40,
    width: 120,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center'
  },
  item2: {
    alignItems: "center",
    backgroundColor: "#ffffff",
    flexGrow: 1,
    margin: 4,
    padding: 20,
    shadowColor: "#000000",
    shadowOpacity: 0.8,
    shadowRadius: 2,
    shadowOffset: {
      height: 1,
      width: 1
    },
    elevation: 4
  },
  text1: {
    color: "#760f9f",
    fontWeight: '200',
  }
});