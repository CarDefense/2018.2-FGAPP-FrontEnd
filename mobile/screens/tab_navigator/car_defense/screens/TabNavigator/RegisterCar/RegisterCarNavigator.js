import React, { Component } from "react";
import {StackNavigator} from 'react-navigation'

import RegisterCar from './RegisterCar'


export default class RegisterCarNavigator extends Component {

    render() {
        return (
            <RegisterCarStackNavigator />
        );
    }
}

const RegisterCarStackNavigator = new StackNavigator({

    RegisterCar: {
        screen: RegisterCar,
        navigationOptions: ({ navigation }) => ({
            header: null,
        }),
    },
})
