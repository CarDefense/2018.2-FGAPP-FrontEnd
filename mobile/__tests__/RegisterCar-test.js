import RegisterCar from '../screens/tab_navigator/car_defense/screens/TabNavigator/RegisterCar/RegisterCar'
import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

test('RegisterCar snapShot', () => {
    const snap = renderer.create(
        <RegisterCar/>
    ).toJSON()
    expect(snap).toMatchSnapshot()
})