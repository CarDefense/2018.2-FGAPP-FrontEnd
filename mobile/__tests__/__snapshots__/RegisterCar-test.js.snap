// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterCar snapShot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#8bd4da",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#8bd4da",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "padding": 30,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "http://cardefense2.eastus.cloudapp.azure.com:8002/media/ssssssssssssssssssssssssssss.png",
          }
        }
        style={
          Object {
            "borderColor": "white",
            "borderRadius": 63,
            "borderWidth": 7,
            "height": 170,
            "marginBottom": 10,
            "width": 170,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontSize": 18,
            "fontWeight": "800",
          }
        }
      >
        Ol√°, 
        ! Cadastre seus carros!
      </Text>
    </View>
  </View>
  <View
    onResponderRelease={[Function]}
    onStartShouldSetResponder={[Function]}
    pointerEvents="auto"
    style={undefined}
  >
    <View
      collapsable={undefined}
      style={
        Object {
          "backgroundColor": "transparent",
          "borderBottomColor": "rgba(0, 0, 0, 0.38)",
          "borderBottomWidth": 0.5,
          "height": 64,
          "paddingBottom": 8,
          "paddingTop": 32,
        }
      }
    >
      <View
        collapsable={undefined}
        style={
          Object {
            "position": "absolute",
            "top": 36,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          collapsable={undefined}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.38)",
              "fontSize": 16,
            }
          }
        >
          Placa
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          autoCapitalize="characters"
          autoCorrect={false}
          disableFullscreenUI={true}
          editable={true}
          enablesReturnKeyAutomatically={true}
          error={undefined}
          maxLength={8}
          onBlur={[Function]}
          onChange={[Function]}
          onChangeText={[Function]}
          onContentSizeChange={[Function]}
          onFocus={[Function]}
          returnKeyType="next"
          selectionColor="white"
          style={
            Array [
              Object {
                "flex": 1,
                "margin": 0,
                "padding": 0,
                "top": 2,
              },
              Object {
                "color": "white",
                "fontSize": 16,
                "height": 24,
                "textAlign": "left",
              },
              undefined,
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
    <View
      collapsable={undefined}
      style={
        Object {
          "flexDirection": "row",
          "height": 8,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "bottom": 0,
              "left": 0,
              "paddingVertical": 4,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            collapsable={undefined}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "rgb(213, 0, 0)",
                "fontSize": 0,
                "opacity": 0,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "bottom": 0,
              "left": 0,
              "paddingVertical": 4,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            collapsable={undefined}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "rgba(0, 0, 0, .38)",
                "fontSize": 12,
                "opacity": 1,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
  <View
    onResponderRelease={[Function]}
    onStartShouldSetResponder={[Function]}
    pointerEvents="auto"
    style={undefined}
  >
    <View
      collapsable={undefined}
      style={
        Object {
          "backgroundColor": "transparent",
          "borderBottomColor": "rgba(0, 0, 0, 0.38)",
          "borderBottomWidth": 0.5,
          "height": 64,
          "paddingBottom": 8,
          "paddingTop": 32,
        }
      }
    >
      <View
        collapsable={undefined}
        style={
          Object {
            "position": "absolute",
            "top": 36,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          collapsable={undefined}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.38)",
              "fontSize": 16,
            }
          }
        >
          Modelo
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          autoCapitalize="sentences"
          autoCorrect={false}
          disableFullscreenUI={true}
          editable={true}
          enablesReturnKeyAutomatically={true}
          onBlur={[Function]}
          onChange={[Function]}
          onChangeText={[Function]}
          onContentSizeChange={[Function]}
          onFocus={[Function]}
          returnKeyType="next"
          selectionColor="white"
          style={
            Array [
              Object {
                "flex": 1,
                "margin": 0,
                "padding": 0,
                "top": 2,
              },
              Object {
                "color": "white",
                "fontSize": 16,
                "height": 24,
                "textAlign": "left",
              },
              undefined,
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
    <View
      collapsable={undefined}
      style={
        Object {
          "flexDirection": "row",
          "height": 8,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "bottom": 0,
              "left": 0,
              "paddingVertical": 4,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            collapsable={undefined}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "rgb(213, 0, 0)",
                "fontSize": 0,
                "opacity": 0,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "bottom": 0,
              "left": 0,
              "paddingVertical": 4,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            collapsable={undefined}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "rgba(0, 0, 0, .38)",
                "fontSize": 12,
                "opacity": 1,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
  <View
    onResponderRelease={[Function]}
    onStartShouldSetResponder={[Function]}
    pointerEvents="auto"
    style={undefined}
  >
    <View
      collapsable={undefined}
      style={
        Object {
          "backgroundColor": "transparent",
          "borderBottomColor": "rgba(0, 0, 0, 0.38)",
          "borderBottomWidth": 0.5,
          "height": 64,
          "paddingBottom": 8,
          "paddingTop": 32,
        }
      }
    >
      <View
        collapsable={undefined}
        style={
          Object {
            "position": "absolute",
            "top": 36,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          collapsable={undefined}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.38)",
              "fontSize": 16,
            }
          }
        >
          Cor
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          autoCapitalize="sentences"
          autoCorrect={false}
          disableFullscreenUI={true}
          editable={true}
          enablesReturnKeyAutomatically={true}
          onBlur={[Function]}
          onChange={[Function]}
          onChangeText={[Function]}
          onContentSizeChange={[Function]}
          onFocus={[Function]}
          returnKeyType="next"
          selectionColor="white"
          style={
            Array [
              Object {
                "flex": 1,
                "margin": 0,
                "padding": 0,
                "top": 2,
              },
              Object {
                "color": "white",
                "fontSize": 16,
                "height": 24,
                "textAlign": "left",
              },
              undefined,
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
    <View
      collapsable={undefined}
      style={
        Object {
          "flexDirection": "row",
          "height": 8,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "bottom": 0,
              "left": 0,
              "paddingVertical": 4,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            collapsable={undefined}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "rgb(213, 0, 0)",
                "fontSize": 0,
                "opacity": 0,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "bottom": 0,
              "left": 0,
              "paddingVertical": 4,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            collapsable={undefined}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "rgba(0, 0, 0, .38)",
                "fontSize": 12,
                "opacity": 1,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": 13,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 15,
          "flexDirection": "row",
          "height": 40,
          "justifyContent": "center",
          "opacity": 1,
          "width": 120,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#8bd4da",
            "fontSize": 15,
            "fontWeight": "800",
          }
        }
      >
        Cadastrar
      </Text>
    </View>
  </View>
</View>
`;
